sys/include/net/sock/udp.h

```diff
@@ -409,8 +465,47 @@ int sock_udp_get_remote(sock_udp_t *sock, sock_udp_ep_t *ep);
  *          the remote of @p sock.
  * @return  -ETIMEDOUT, if @p timeout expired.
  */
-ssize_t sock_udp_recv(sock_udp_t *sock, void *data, size_t max_len,
-                      uint32_t timeout, sock_udp_ep_t *remote);
+ssize_t sock_udp_recv_aux(sock_udp_t *sock, void *data, size_t max_len,
```

Comment:
If I read the diff right, this is a 1:1 copy of the recv function (a with all analogous ones).

Both in terms of maintainability (changing one and forgetting the other) and of readability ("Can't look at them side-by-side ... I'll just hope they're really the same"), I think it'd be better to deduplicate here, even if it means that formally, not all parameters are documented. (We don't have checks for that, do we?)

The whole `sock_udp_recv_aux` doc might say (emphasis indicating additions)

```
@brief   Receives a UDP message from a remote end point *with additional data*

*Preconditions, parametes and return value are as in @ref sock_udp_recv, with
one additional parameter:*

@param[out] aux      Auxiliary data about the received datagram.
                     May be `NULL`, if it is not required by the application.

                     *Fields in @p aux are populated as requested in its
                     `flags` field using @ref SOCK_AUX_GET_LOCAL or
                     similar flags.
```

(If the flags enum can be referenced even though it is anonymous, it might make a better reference target than picking the first example).

Reply:
> (We don't have checks for that, do we?)

With the "new" stable Doxygen version, this would fail. A lot of failures like this are all over the code base, which is currently blocking the transition to thew new version. If I would deduplicate this, I expect this to be duplicated again once we finally update :-/

Reply:
Wait, I think it only requires that there is a @brief for each function. Let me check, if Doxygen complains.

Reply:
Yes, it does :-/

`RIOT/sys/include/net/sock/udp.h:435: warning: The following parameters of sock_udp_recv_aux(sock_udp_t *sock, void *data, size_t max_len, uint32_t timeout, sock_udp_ep_t *remote, sock_udp_aux_rx_t *aux) are not documented:` ...

Reply:
Then duplication is fine with me here (maybe I'll find the spoons to pluck that crow with doxygen); a note on equivalence a la

```
This is functionally equivalent with calling @ref sock_dtls_recv_aux with `NULL` as the @p aux parameter.
```

and

```
When @p aux is not needed, @ref sock_dtls_recv is provided as a shortcut with otherwise identical parameters.
```

might make the navigation easier to navigate.

(On the general review level, I'm still waiting for a more qualified fundamentals review.)